xdsl/pattern_rewriter.py

```diff
@@ -471,6 +473,58 @@ def move_region_contents_to_new_regions(self, region: Region) -> Region:
             )
         return Rewriter.move_region_contents_to_new_regions(region)
 
+    def inline_region_before(self, region: Region, target: Block) -> None:
+        """Move the region blocks to an existing region."""
+        self.has_done_action = True
+        if not self._can_modify_region(region):
+            raise Exception(
+                "Cannot move regions that are not children of the matched operation"
+            )
```

Comment:
I guess we could get rid of some code duplication here by making `_can_modify_region` something like `_assert_can_modify_region` and make it raise the Exception itself (the exception seems to be a pretty generic string in all cases anyways).

Reply:
Yeah that's a good idea, could you please file an issue? Seems like a good starter bug for someone

Reply:
Just so you know, these restrictions will definitely be removed in like 1-2 PRs, so we won't need to change that at all!